src/presentation/http/router/noteSettings.ts

```diff
@@ -173,6 +174,43 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(team);
   });
 
+  /**
+   * Generates a new invitation hash for a specific note
+   */
+  fastify.patch<{
+    Params: {
+      notePublicId: NotePublicId;
+    },
+    Reply: InvitationHash,
+  }>('/:notePublicId/invitation-hash', {
+    config: {
+      policy: [
+        'authRequired',
+        'userInTeam',
+      ],
+    },
+    schema: {
+      params: {
+        notePublicId: {
+          $ref: 'NoteSchema#/properties/id',
+        },
+      },
+    },
```

Comment:
and specify response schema as well

https://fastify.dev/docs/latest/Guides/Getting-Started/#serialize-your-data